samples/create-and-edit-geometries/src/main/java/com/esri/arcgismaps/sample/createandeditgeometries/components/CreateAndEditGeometriesViewModel.kt

```diff
@@ -0,0 +1,147 @@
+
+    // create a MapViewProxy that will be used to identify features in the MapView and set the viewpoint
+    val mapViewProxy = MapViewProxy()
+
+    // create a graphic, graphic overlay, and geometry editor
+    private var identifiedGraphic = Graphic()
+    val graphicsOverlay = GraphicsOverlay()
+    val geometryEditor = GeometryEditor()
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    title = "Failed to load map",
+                    description = error.message.toString()
+                )
+            }
+        }
+    }
+
+    /**
+     * Starts the GeometryEditor using the selected [GeometryType].
+     */
+    fun startEditor(selectedGeometry: GeometryType) {
+        if (!geometryEditor.isStarted.value) {
+            geometryEditor.start(selectedGeometry)
+        }
+    }
+
+    /**
+     * Stops the GeometryEditor and updates the identified graphic or calls [createGraphic].
+     */
+    fun stopEditor() {
+        if (identifiedGraphic.geometry != null) {
+            identifiedGraphic.geometry = geometryEditor.stop()
+            identifiedGraphic.isSelected = false
+        } else if (geometryEditor.isStarted.value) {
+            createGraphic()
+        }
+    }
+
+    /**
+     * Creates a graphic from the geometry and adds it to the GraphicsOverlay.
+     */
+    private fun createGraphic() {
+        val geometry = geometryEditor.stop()
+            ?: return messageDialogVM.showMessageDialog(
+                title = "Error!",
+                description = "Error stopping editing session"
+            )
+        val graphic = Graphic(geometry)
+
+        when (geometry) {
+            is Point, is Multipoint -> graphic.symbol = GeometryEditorStyle().vertexSymbol
+            is Polyline -> graphic.symbol = GeometryEditorStyle().lineSymbol
+            is Polygon -> graphic.symbol = GeometryEditorStyle().fillSymbol
+            else -> {}
+        }
+        graphicsOverlay.graphics.add(graphic)
+        graphic.isSelected = false
+    }
+
+    /**
+     * Identifies the graphic at the tapped screen coordinate in the provided [singleTapConfirmedEvent]
+     * and starts the GeometryEditor using the identified graphic's geometry. Hide the BottomSheet on
+     * [singleTapConfirmedEvent].
+     */
+    fun identify(singleTapConfirmedEvent: SingleTapConfirmedEvent) {
+        viewModelScope.launch {
+            val graphicsResult = mapViewProxy.identifyGraphicsOverlays(
+                screenCoordinate = singleTapConfirmedEvent.screenCoordinate,
+                tolerance = 10.0.dp,
+                returnPopupsOnly = false
+            ).getOrNull()
+
+            if (!geometryEditor.isStarted.value) {
+                if (graphicsResult != null) {
+                    if (graphicsResult.isNotEmpty()) {
+                        identifiedGraphic = graphicsResult[0].graphics[0]
```

Comment:
```suggestion
                        identifiedGraphic = graphicsResult.first().graphics.first()
```

kotlin gives us this nice bit of syntactic sugar